py/desispec/exposure_qa.py

```diff
@@ -0,0 +1,377 @@
+    if not os.path.isfile(fibermap_filename) :
+        log.warning("no {}".format(fibermap_filename))
+        return None , None
+
+    fibermap = read_fibermap(fibermap_filename)
+    petal_locs=np.unique(fibermap["PETAL_LOC"])
+
+    fiberqa_table = Table()
+    for k in ['TARGETID', 'PETAL_LOC', 'DEVICE_LOC', 'LOCATION', 'FIBER', 'TARGET_RA', 'TARGET_DEC',
+              'FIBER_X', 'FIBER_Y', 'DELTA_X', 'DELTA_Y', 'EBV'] :
+        fiberqa_table[k]=fibermap[k]
+
+    fiberqa_table['QAFIBERSTATUS']=fibermap['FIBERSTATUS']  # copy because content will be different
+
+    fiberqa_table.meta["NIGHT"]=night
+    fiberqa_table.meta["EXPID"]=expid
+    #fiberqa_table.meta["PRODDIR"]=specprod_dir
+
+
+    x_mm  = fibermap["FIBER_X"]
+    y_mm  = fibermap["FIBER_Y"]
+    dx_mm = fibermap["DELTA_X"]
+    dy_mm = fibermap["DELTA_Y"]
+
+    nan_positions = np.isnan(x_mm)|np.isnan(y_mm)
+    x_mm[nan_positions]=0.
+    y_mm[nan_positions]=0.
+
+    nan_positions |= np.isnan(dx_mm)|np.isnan(dy_mm)
+    dx_mm[nan_positions]=0.
+    dy_mm[nan_positions]=0.
+
+    # nan = no data
+    fiberqa_table['QAFIBERSTATUS'][nan_positions] |= fibermask.mask('MISSINGPOSITION')
+
+    dist_mm = np.sqrt(dx_mm**2+dy_mm**2)
+    poorposition=(dist_mm>qa_params["max_fiber_offset_mm"])
+    fiberqa_table['QAFIBERSTATUS'][poorposition] |= fibermask.mask('POORPOSITION')
+
+    petal_tsnr2=np.zeros(10)
+    worst_rdnoise = 0
+
+    fiberqa_table["EFFTIME_SPEC"]=np.zeros(fiberqa_table["TARGETID"].size)
+
+    petalqa_table = Table()
+    npetal=10
+    petalqa_table["PETAL_LOC"]=np.arange(npetal)
+    petalqa_table["WORSTREADNOISE"]=np.zeros(npetal,dtype=float)
+    petalqa_table["NGOODPOS"]=np.zeros(npetal,dtype=int)
+    petalqa_table["NGOODFIB"]=np.zeros(npetal,dtype=int)
+    petalqa_table["NSTDSTAR"]=np.zeros(npetal,dtype=int)
+    petalqa_table["STARRMS"]=np.zeros(npetal,dtype=float)
+    petalqa_table["TSNR2FRA"]=np.zeros(npetal,dtype=float)
+    petalqa_table["EFFTIME_SPEC"]=np.zeros(npetal,dtype=float)
+    petalqa_table["NCFRAME"]=np.zeros(npetal,dtype=int)
+    petalqa_table["BSKYTHRURMS"]=np.zeros(npetal,dtype=float)
+    petalqa_table["BSKYCHI2PDF"]=np.zeros(npetal,dtype=float)
+    petalqa_table["RSKYTHRURMS"]=np.zeros(npetal,dtype=float)
+    petalqa_table["RSKYCHI2PDF"]=np.zeros(npetal,dtype=float)
+    petalqa_table["ZSKYTHRURMS"]=np.zeros(npetal,dtype=float)
+    petalqa_table["ZSKYCHI2PDF"]=np.zeros(npetal,dtype=float)
+    petalqa_table["BTHRUFRAC"]=np.zeros(npetal,dtype=float)
+    petalqa_table["RTHRUFRAC"]=np.zeros(npetal,dtype=float)
+    petalqa_table["ZTHRUFRAC"]=np.zeros(npetal,dtype=float)
+
+    # need to add things
+
+    frame_header = None
+    fibermap_header = None
+
+    # EFFTIME
+    goaltype="dark"
+    if fibermap_header is not None and "GOALTYPE" in fibermap_header :
+        goaltype=fibermap_header["GOALTYPE"]
+    else :
+        log.warning("no GOALTYPE info, assume 'dark'")
```

Comment:
Just before this if/else you set `fibermap_header = None`, which means that it will always not find GOALTYPE and default to dark.  I think you mean to get the `fibermap_header` from the fibermap file instead?  Note that below there is another check on `fibermap_header is None` in which case it is ready from the cframe file, which also might be a leftover.

In particular the fibermap_header is then read from the cframe and its GOALTYPE is is propagated to the output QA.

Reply:
Thanks for finding this one. I now use directly fibermap.meta.